src/os/CMakeLists.txt

```diff
@@ -122,18 +128,20 @@ if(HAVE_LIBAIO)
 endif()
 
 if(WITH_SPDK)
+  include(BuildDPDK)
   include_directories("${CMAKE_SOURCE_DIR}/src/spdk/include")
   add_custom_target(build_spdk
     COMMAND
-    $(MAKE) DPDK_INC_DIR=${DPDK_INCLUDE_DIR}
-    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/spdk/lib
+    $(MAKE) DPDK_DIR=${CMAKE_SOURCE_DIR}/src/spdk/dpdk/build
+    DEPENDS build_dpdk
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/spdk
     COMMENT "spdk building")
   # TODO: should use add_library(spdk INTERFACE IMPORTED) instead in new cmake,
   # if INTERFACE is supported.
   foreach(lib nvme log env_dpdk util)
     add_library(spdk_${lib} STATIC IMPORTED)
-    add_dependencies(spdk_${lib} build_spdk)
-    target_link_libraries(os PRIVATE spdk_${lib})
+    add_dependencies(spdk_${lib} build_dpdk build_spdk)
```

Comment:
again, as specified in above `add_custom_target()` command, `build_spdk` depends on `build_dpdk` . so no need to add this dependecy, am i right?

Reply:
yeah, you are right ... I removed this build_dpdk, and it works ok. Thanks!